### DISCUSSION TITLE
[Feature Request] Allow WebAssembly for creating GitHub Actions

### DISCUSSION THREAD

**Discussion by Victor-N-Suadicani** [community/community#39920] [Actions]
### Select Topic Area

Product Feedback

### Body

At the time of writing, the [About custom actions docs](https://docs.github.com/en/actions/creating-actions/about-custom-actions) specify that you can either write custom actions as a Docker container or as JavaScript. It also says that JavaScript actions "executes faster than a Docker container action".

It would be great if you could write custom actions using WebAssembly (WASM) in the same way as when you use JavaScript. This would allow the same efficiency benefits (not having to setup the whole Docker container) while allowing a much wider range of languages to be used for GitHub Actions. This would enable many more actions as you could draw on (basically) any language you want and thus have access to a wider ecosystem than just JavaScript.

WebAssembly also has the potential to be faster to execute than JavaScript. though this isn't the primary reason for this feature request.

**Comment by ossanna16** [comment]
Hi there @Victor-N-Suadicani and welcome to our community! Thank you for the feedback!
